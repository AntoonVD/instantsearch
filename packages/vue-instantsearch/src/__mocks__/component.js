import suit from '../suit';

export default {
  data() {
    return {
      state: {
        refine: jest.fn(),
      },
    };
  },
  methods: {
    suit(...args) {
      return suit(this.widgetName, ...args);
    },
  },
};
