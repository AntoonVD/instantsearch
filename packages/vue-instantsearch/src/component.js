import suit from './suit.js';

export default {
  inject: ['_searchStore', '_instance'],
  props: {
    searchStore: {
      type: Object,
      default() {
        if (typeof this._searchStore !== 'object') {
          const tag = this.$options._componentTag;
          throw new TypeError(
            `It looks like you forgot to wrap your Algolia search component 
            "<${tag}>" inside of an "<ais-index>" component. You can also pass a 
            search store as a prop to your component.`
          );
        }
        return this._searchStore;
      },
    },
    instance: {
      type: Object,
      default() {
        if (typeof this._instance !== 'object') {
          const tag = this.$options._componentTag;
          throw new TypeError(
            `It looks like you forgot to wrap your Algolia search component 
            "<${tag}>" inside of an "<ais-index>" component. You can also pass a 
            search store as a prop to your component.`
          );
        }
        return this._instance;
      },
    },
    classNames: {
      type: Object,
      default() {
        return {};
      },
    },
  },
  data() {
    return {
      state: {},
    };
  },
  beforeCreate() {
    let source = this; // eslint-disable-line consistent-this
    const store = '_searchStore';
    const instance = '_instance';

    while (source) {
      if (source._provided && store in source._provided) {
        break;
      }
      if (source._provided && instance in source._provided) {
        break;
      }
      source = source.$parent;
    }

    if (!source) {
      if (!this._provided) {
        this._provided = {};
      }

      this._provided[store] = undefined;
      this._provided[instance] = undefined;
    }
  },
  created() {
    if (this.connector) {
      this.widgetFactory = this.connector(this.updateData, () => {});
    }
    if (this.widgetFactory) {
      this.widget = this.widgetFactory(this.widgetParams);
      this.instance.addWidget(this.widget);
    }
  },
  beforeDestroy() {
    // todo: remove dispose flub
    if (this.widget && this.widget.dispose) {
      this.instance.removeWidget(this.widget);
    }
  },
  watch: {
    widgetParams(newVal) {
      if (this.widget) {
        const oldWidget = this.widget;
        this.widget = this.widgetFactory(newVal);

        this.instance.addWidget(this.widget);
        // todo: remove dispose flub
        if (oldWidget.dispose) {
          this.instance.removeWidget(oldWidget);
        }
      }
    },
  },
  methods: {
    suit(...args) {
      return suit(this.widgetName, ...args);
    },
    updateData(state = {}) {
      this.state = state;
    },
  },
};
