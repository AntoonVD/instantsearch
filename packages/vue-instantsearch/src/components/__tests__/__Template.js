import { mount } from '@vue/test-utils';
import Template from '../__Template.vue';
import { __setState } from '../../component';
jest.mock('../../component');

it('renders correctly', () => {
  __setState({
    hits: ['yo', 'how', 'are', 'you', 'doing', '?'],
  });
  const wrapper = mount(Template);
  expect(wrapper.html()).toMatchSnapshot();
});

// ☑️ add another rendering test if it's different given the propsData

it('behaves correctly', () => {
  const wrapper = mount(Template);
  const button = wrapper.find('button');
  button.trigger('click');
  expect(wrapper.vm.state.refine).toHaveBeenLastCalledWith('hi');
});
