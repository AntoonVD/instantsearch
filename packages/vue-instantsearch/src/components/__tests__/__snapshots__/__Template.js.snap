// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Template html should render correctly 1`] = `

<div>
  <button>
  </button>
  <pre>
    {
  "hits": []
}
  </pre>
</div>

`;

exports[`Template html should render correctly when \`hits\` is different 1`] = `

<div>
  <button>
  </button>
  <pre>
    {
  "hits": [
    "yo",
    "how",
    "are",
    "you",
    "doing",
    "?"
  ]
}
  </pre>
</div>

`;
